lib/fcm/fcm.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+import '../base/firebase.dart';
+import 'message.dart';
+
+/// FCM Option defined here: https://firebase.google.com/docs/cloud-messaging/send-message#rest
+/// Don't know why you should overwrite these settings but you can.
+class FCMConfig {
+  static const scheme = 'https';
+  static const host = 'fcm.googleapis.com';
+  static const method = 'POST';
+  static const bool keepAlive = false;
+  final headers = <String,String>{};
+  String project_id;
+
+  FCMConfig(this.project_id);
+
+  String get path => '/v1/projects/$project_id/messages:send';
+}
+
+class FCM {
+  /* Singleton instance */
+  static FCM _instance;
+
+  static bool get initialized => _instance != null;
+
+  static FCM initialize(
+      {Firebase firebase, FCMConfig fcmConfig}) {
+    assert(!initialized,
+    'Firestore global instance is already initialized. Do not call this twice or create a local instance via FCM()');
+
+    _instance = FCM(
+        firebase: firebase ?? Firebase.instance, fcmConfig: fcmConfig);
+    return _instance;
+  }
+
+  static FCM get instance {
+    assert(initialized,
+    "Firestore hasn't been initialized. Call Firestore.initialize() before using this global instance. Alternatively, create a local instance via Firestore() and use that.");
+
+    return _instance;
+  }
+
+  /* Instance interface */
+  final Firebase firebase;
+  final FCMConfig fcmConfig;
+
+  FCM({@required this.firebase, @required this.fcmConfig})
+      : assert(firebase != null || Firebase.initialized,
+  'Firebase global instance not initialized, run Firebase.initialize().\nAlternatively, provide a local instance via Firestore.initialize(firebase: <firebase instance>)'),
+        assert(fcmConfig != null, 'Firebase Cloud Messaging configuration is missing.');
+
+  /// https://firebase.google.com/docs/cloud-messaging/send-message#send-messages-to-specific-devices
+  ///
+  /// Throws a V1FcmError if the requests fails or server replies an error.
+  ///
+  /// Returns ID string in format projects/{project_id}/messages/{message_id}
+  /// if the request is successful.
+  /// Example: "projects/myproject-b5ae1/messages/0:1500415314455276%31bd1c9631bd1c96"
```

Comment:
> projects/myproject-b5ae1/messages/0:1500415314455276%31bd1c9631bd1c96
Example: "projects/myproject-b5ae1/messages/0:1500415314455276%31bd1c9631bd1c96"

What is this? Change this. Alternatively, provide the example directly in the doc here

Reply:
This is exactly the doc, the method returns an `ID string`, it uniquely identifies the message. I've updated the comment with this information, let me know if it helps.